packages/react-stand-in/src/createClassProxy.js

```diff
@@ -48,12 +47,42 @@ function createClassProxy(InitialComponent, proxyKey, wrapResult = identity) {
       ? CurrentComponent(this.props, this.context)
       : CurrentComponent.prototype.render.call(this)
 
-    return wrapResult(result, this.props, this.context)
+    return wrapResult(result)
   }
 
-  let ProxyComponent = proxyClassCreator(InitialParent, postConstructionAction)
+  let ProxyComponent
+  let ActualProxyComponent = null
+
+  if (!isFunctionalComponent) {
+    ProxyComponent = proxyClassCreator(InitialComponent, postConstructionAction)
+    ProxyComponent.prototype.render = proxiedRender
+
+    ActualProxyComponent = ProxyComponent
+  } else {
+    // This function only gets called for the initial mount. The actual
+    // rendered component instance will be the return value.
+    ProxyComponent = function(props, context) {
+      const result = CurrentComponent(props, context)
+
+      // This is a Relay-style container constructor. We can't do the prototype-
+      // style wrapping for this as we do elsewhere, so just we just pass it
+      // through as-is.
+      if (isReactIndeterminateResult(result)) {
+        ActualProxyComponent = null
+        return result
```

Comment:
Technically we could use the older ways of proxying this here. Maybe this should be a `FIXME` or a `TODO`. I've never seen this pattern get used outside of Relay, and you can't really hot-reload Relay containers in a useful way anyway, because the queries are static.